### ISSUE TITLE
Backspace is wrongly transformed into an insertion in TestBackspaceInParenthesizedDeconstructionDeclaration

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#15448]
Although the suggestion mode provider is only triggered on insertions, it is triggered by the second backspace. I need to investigate the root cause.

```VB
        <WpfFact, Trait(Traits.Feature, Traits.Features.Completion)>
        Public Async Function TestBackspaceInParenthesizedDeconstructionDeclaration() As Task
            Using state = TestState.CreateCSharpTestState(
                  <Document><![CDATA[
class Variable
{
    public void Foo()
    {
       (var as$$)
    }
}]]></Document>)

                state.Workspace.Options = state.Workspace.Options.WithChangedOption(
                    CompletionOptions.TriggerOnDeletion, LanguageNames.CSharp, True)

                state.SendBackspace()
                ' This completion is hard-selected because the suggestion mode never triggers on backspace
                ' See issue https://github.com/dotnet/roslyn/issues/15302
                Await state.AssertSelectedCompletionItem(displayText:="as", isHardSelected:=True)

                state.SendTypeChars(", var as")
                state.SendBackspace() ' <------- HERE
                Await state.AssertSelectedCompletionItem(displayText:="as", isSoftSelected:=True)

                state.SendReturn()
                Await state.AssertNoCompletionSession()
                Assert.Contains("(var as, var a", state.GetLineTextFromCaretPosition(), StringComparison.Ordinal)
            End Using
        End Function
```

**Comment by jcouv** [comment]
From my investigation so far, it seems the difference in behavior is because the first `SendBackspace` is called on its own, whereas the second one is called after `SendTypeChars`. In the second case, the trigger gets captured into some session object and the suggestion mode provider is only invoked with the first letter of the last word typed, but never with the backspace.

**Comment by jcouv** [comment]
@rchande From our discussion, it sounds like this is by design. When the session starts, a trigger is captured, and that trigger is what determines completion logic even if a backspace is later typed.
We can close if that is indeed expected.
